batcher/aligned/src/main.rs

```diff
@@ -107,20 +109,85 @@ pub struct SubmitArgs {
     #[arg(
         name = "The working network's name",
         long = "network",
         default_value = "devnet"
     )]
-    network: NetworkArg,
+    network: Network,
+}
+
+#[derive(Args, Debug)]
+#[group(required = false, multiple = false)]
+pub struct FeeType {
+    #[arg(
+        name = "Max Fee (ether)",
+        long = "max_fee",
+        help = "Specifies the maximum fee (`max_fee`) the user is willing to pay for the submitted proof, in Ether."
+    )]
+    max_fee: Option<String>, // String because U256 expects hex
+    #[arg(
+        name = "NUMBER_PROOFS_IN_BATCH",
+        long = "custom_fee_estimate",
+        help = "Specifies a `max_fee` equivalent to the cost of paying (1 proof / `num_proofs_in_batch`) allowing the user to estimate a `max_fee` precisely based on the `number_proofs_in_batch`."
+    )]
+    custom_fee_estimate: Option<usize>,
+    #[arg(
+        long = "instant_fee_estimate",
+        help = "Specifies a `max_fee` that covers the cost of paying for the entire batch, ensuring the proof is included instantly."
+    )]
+    instant_fee_estimate: bool,
+    #[arg(
+        long = "default_fee_estimate",
+        help = "Specifies a default `max_fee` based on the cost of one proof within a batch of 16 proofs, providing a standard fee for batch inclusion."
+    )]
+    default_fee_estimate: bool,
+}
+
+impl SubmitArgs {
+    async fn get_max_fee(&self) -> Result<U256, AlignedError> {
+        if let Some(max_fee) = &self.fee_type.max_fee {
+            if !max_fee.ends_with("ether") {
+                error!("`max_fee` should be in the format XX.XXether");
+                Err(SubmitError::EthereumProviderError(
+                    "Error while parsing `max_fee`".to_string(),
+                ))?
+            }
+
+            let max_fee_ether = max_fee.replace("ether", "");
+            return Ok(parse_ether(max_fee_ether).map_err(|e| {
+                SubmitError::EthereumProviderError(format!("Error while parsing `max_fee`: {}", e))
+            })?);
+        }
+
+        if let Some(number_proofs_in_batch) = &self.fee_type.custom_fee_estimate {
+            return estimate_fee(
+                &self.eth_rpc_url,
+                FeeEstimateType::Custom(*number_proofs_in_batch),
+            )
+            .await
+            .map_err(AlignedError::FeeEstimateError);
+        }
+
+        if self.fee_type.instant_fee_estimate {
+            return estimate_fee(&self.eth_rpc_url, FeeEstimateType::Instant)
+                .await
+                .map_err(AlignedError::FeeEstimateError);
+        }
+
+        if self.fee_type.default_fee_estimate {
+            return estimate_fee(&self.eth_rpc_url, FeeEstimateType::Default)
+                .await
+                .map_err(AlignedError::FeeEstimateError);
+        }
+
+        Ok(U256::from_dec_str("13000000000000")
+            .map_err(|e| SubmitError::GenericError(e.to_string()))?) // 13_000 gas per proof * 100 gwei gas price (upper bound)
```

Comment:
here we should use the default_batch_size fee, this is why we created it, to be used as default